packages/grid/_modules_/grid/utils/domUtils.ts

```diff
@@ -62,7 +62,7 @@ export function getGridColumnHeaderElement(root: Element, field: string) {
 
 export function getGridRowElement(root: Element, id: GridRowId) {
   return root.querySelector(
-    `:scope .${GRID_ROW_CSS_CLASS}[data-id="${escapeOperandAttributeSelector(String(id))}"]`,
+    `.${GRID_ROW_CSS_CLASS}[data-id="${escapeOperandAttributeSelector(String(id))}"]`,
```

Comment:
we already have an `escapeRegExp`, can't we reuse it?

Reply:
I don't understand regexp and attribute selectors are unrelated.

Reply:
hence the question `can't we reuse it?` if they both result in `\'` or similar we might not need a new method.
Also removing `:scope` will now allow the header row as a result

Reply:
OK, I will keep two different methods because the concepts are unrelated.

Regarding `:scope`, do you have a simple codesandbox to illustrate the issue?

Reply:
I have reverted the change around :scope as it doesn't need to be in this PR, and it also doesn't seem to be that trivial. Move to #2115.